src/components/Datagrid/FullScreenContainer.tsx

```diff
@@ -50,14 +50,36 @@ const useAnimationStyles = (isOpen: boolean, duration: number) => {
   };
 };
 
-interface FullScreenContainerProps {
-  open?: boolean;
-  className?: string;
-}
+const useEscapeKey = (callback?: () => void) => {
+  useEffect(() => {
+    const handleEscapeKey = (event: KeyboardEvent) => {
+      if (event.code === "Escape") {
+        callback();
+      }
+    };
 
-const Portal = ({ className, children, open }) => {
+    document.addEventListener("keydown", handleEscapeKey);
+    return () => document.removeEventListener("keydown", handleEscapeKey);
+  }, [callback]);
+};
+
+type FullScreenContainerProps = FC<
+  PropsWithChildren<{
+    open?: boolean;
+    className?: string;
+    onEscapePress?: () => void;
+  }>
+>;
+
+const Portal: FullScreenContainerProps = ({
+  className,
+  children,
+  open,
+  onEscapePress,
+}) => {
   const { delayedState: delayedOpen, duration } = useDelayedState(open);
   const styles = useAnimationStyles(open, duration);
+  useEscapeKey(onEscapePress);
```

Comment:
there is already a hook `useFullScreenMode` wich is responsible for opening / closing full-screen mode, it makes more sense to use `useEscapeKey` in there. You will avoid additional props here, the only thing this component needs is information if the portal is open/closed while this state can come from a variety of places (abstraction).

So i suggest to move `useEscapeKey` to either separate file or to `useFullScreenMode`, then using it within `useFullScreenMode` so you supossed to have just two files modified

Reply:
fixed - please take a look 😄